app/assets/javascripts/Components/Modals/create_modify_annotation_panel_modal.jsx

```diff
@@ -0,0 +1,132 @@
+import React from "react";
+import Modal from "react-modal";
+
+class CreateModifyAnnotationPanel extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      content: this.props.content,
+      category_id: this.props.category_id,
+    };
+  }
+
+  componentDidMount() {
+    Modal.setAppElement("body");
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps !== this.props) {
+      this.setState({
+        content: this.props.content,
+        category_id: this.props.category_id,
+      });
+    }
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    this.props.onSubmit({ ...this.state });
+    this.setState({ content: "", category_id: "" });
+  };
+
+  handleChange = (event) => {
+    let name = event.target.name;
+    let val = event.target.value;
+    this.setState({ [name]: val });
+  };
+
+  checkCriterion = (event) => {
+    event.preventDefault();
+    this.handleChange(event);
+    let val = event.target.value;
+    let categories_with_criteria = this.props.categories
+      .filter((category) => category.flexible_criterion_id !== null)
+      .map((category) => category.id);
+
+    if (categories_with_criteria.includes(parseInt(val))) {
```

Comment:
Use `parseInt(val, 10)` here to specify the base (to avoid problems described https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#Description).